cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/jmx/Nodetool.java

```diff
@@ -0,0 +1,118 @@
+        this.jmxConnect = jmxConnect;
+    }
+
+    public boolean isNativeTransportRunning() {
+        return jmxConnect.getStorageServiceProxy().isNativeTransportRunning();
+    }
+
+    public boolean isRPCServerRunning() {
+        return jmxConnect.getStorageServiceProxy().isRPCServerRunning();
+    }
+
+    /**
+     * Whether the node has joined the ring.
+     */
+    public boolean isJoined() {
+        return jmxConnect.getStorageServiceProxy().isJoined();
+    }
+
+    /**
+     * to determine if gossip is disabled
+     */
+    public boolean isGossipInitialized() {
+        return jmxConnect.getStorageServiceProxy().isInitialized();
+    }
+
+    /**
+     * whether gossip is running or not
+     */
+    public boolean isGossipRunning() {
+        return jmxConnect.getStorageServiceProxy().isGossipRunning();
+    }
+
+    /**
+     * Get the operational mode.
+     * Valid operation modes are:
+     * <ul>
+     *     <li>STARTING</li>
+     *     <li>NORMAL</li>
+     *     <li>JOINING</li>
+     *     <li>LEAVING</li>
+     *     <li>DECOMMISSIONED</li>
+     *     <li>MOVING</li>
+     *     <li>DRAINING</li>
+     *     <li>DRAINED</li>
+     * </ul>
+     */
+    public String getOperationMode() {
+        return jmxConnect.getStorageServiceProxy().getOperationMode();
+    }
+
+    public long getUptimeInMillis() {
+        return jmxConnect.getRuntimeProxy().getUptime();
+    }
+
+    public String getHostID() {
+        return jmxConnect.getStorageServiceProxy().getLocalHostId();
+    }
+
+    public int getTokenCount() {
+        return jmxConnect.getStorageServiceProxy().getTokens().size();
+    }
+
+    public String getVersion() {
+        return jmxConnect.getStorageServiceProxy().getReleaseVersion();
+    }
+
+    public String getClusterName() {
+        return jmxConnect.getStorageServiceProxy().getClusterName();
+    }
+
+    public String getDataCenter(String endpoint) throws UnknownHostException {
+        return jmxConnect.getEndpointSnitchInfoProxy().getDatacenter(endpoint);
+    }
+
+    public String getRack(String endpoint) throws UnknownHostException {
+        return jmxConnect.getEndpointSnitchInfoProxy().getRack(endpoint);
+    }
+
+    public String getEndpoint()
+    {
+        // Try to find the endpoint using the local token, doing so in a crazy manner
+        // to maintain backwards compatibility with the MBean interface
+        String stringToken = jmxConnect.getStorageServiceProxy().getTokens().get(0);
+        Map<String, String> tokenToEndpoint = jmxConnect.getStorageServiceProxy().getTokenToEndpointMap();
+
+        for (Map.Entry<String, String> pair : tokenToEndpoint.entrySet())
+        {
+            if (pair.getKey().equals(stringToken))
+            {
+                return pair.getValue();
+            }
+        }
+
+        throw new RuntimeException("Could not find myself in the endpoint list, something is very wrong!  Is the Cassandra node fully started?");
```

Comment:
A more specific exception would help in being able to communicate the error scenario more effectively.